lib/helper.c

```diff
@@ -326,8 +329,18 @@ int fuse_main_real(int argc, char *argv[], const struct fuse_operations *op,
 
 	if (opts.singlethread)
 		res = fuse_loop(fuse);
-	else
-		res = fuse_loop_mt(fuse, opts.clone_fd);
+	else {
+		struct fuse_loop_config loop_config;
+		loop_config.clone_fd = opts.clone_fd;
+		if (opts.max_idle_threads == 0) {
```

Comment:
Just assign the default *prior* to calling `fuse_parse_cmdline` (line 279), then you don't need the `if` condition. It actually looks like there is a bug there already, in that `opts` is not explicitly zerod. It should be:

```
 	struct fuse_cmdline_opts opts;
 	int res;
 
        memset(&opts, 0, sizeof(opts));
        opts.max_idle_threads = 10;        
 	if (fuse_parse_cmdline(&args, &opts) != 0)
return 1;
```